From 65abb7976c2c626e241954a266aea1f888b4b97c Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 17 May 2019 12:16:41 +0530
Subject: [PATCH 0187/1921] octeontx2-af: Use proper error message

Bail out with proper message when ntuple filter cannot
be installed because of uninitialized interface.

Change-Id: Ia7b943b68951d2d859d23b064114373a4dc5de98
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/9136
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
index 1ffc28448502..0f99a70c3962 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
@@ -931,7 +931,8 @@ int rvu_mbox_handler_npc_install_flow(struct rvu *rvu,
 	pfvf = rvu_get_pfvf(rvu, target);
 
 	if (!req->default_rule && !pfvf->def_rule) {
-		dev_err(rvu->dev, "default rule should be installed first\n");
+		dev_err(rvu->dev,
+			"Cannot install rule target interface uninitialized\n");
 		return -EINVAL;
 	}
 
-- 
2.31.1

